l10n_ch_delivery_carrier_label_quickpac/models/stock.py

```diff
@@ -0,0 +1,79 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+import base64
+
+from odoo import _, api, models
+from odoo.exceptions import UserError
+
+from ..quickpac.web_service import QuickpacWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    @api.multi
+    @api.onchange("carrier_id")
+    def onchange_carrier_id(self):
+        for picking in self:
+            partner = picking.partner_id
+            delivery_type = picking.carrier_id.delivery_type
+            if delivery_type != "quickpac" or not partner:
+                continue
+            service = QuickpacWebService(picking.company_id)
+            if not service.is_deliverable_zipcode(partner.zip):
+                raise UserError(
+                    _("This partner is undeliverable (invalid zipcode).")
+                )
```

Comment:
I think it is not a good idea to block the selection of the delivery method at this point.
A given ZIP can be invalid today, but valid in 3 days (or valid today and invalid in 3 days), so if a user sets the delivery method some days before the delivery, IMO they should be able to select it (which can be a default value or set from the sale order BTW) whatever. Only the generation of the label should actually be prevented if the ZIP is not valid.

Reply:
You're right, i'll remove that part. The error will raise if needed later on after all

Reply:
Thanks!